libvast/src/system/node.cpp

```diff
@@ -481,10 +486,12 @@ node_state::spawn_command(const invocation& inv,
       rp.deliver(component.error());
       return caf::make_message(std::move(component.error()));
     }
-    self->monitor(*component);
-    auto okay = self->state.registry.add(*component, std::move(comp_type),
-                                         std::move(label));
-    VAST_ASSERT(okay);
+    if (self->state.registry.add(*component, std::move(comp_type), label))
```

Comment:
Both add and remove in the component registry should probably be marked nodiscard.

Reply:
Yep, added.